### PR TITLE
feat: support kubebuilder

### PR DESCRIPTION
**Pull request by AlinsRan**
### Type of change:

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:

* Can use kuberbuider to automatically update CRD resources.
* No longer need to maintain CRD resources.
    * Prone to errors and omissions.

use: 
* make manifests

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by lingsamuel** [comment]
/hold
https://github.com/apache/apisix-ingress-controller/issues/515#issuecomment-1235265365 Careful verification is required that the CRD checking rules have been properly ported to the code base. 
Also, old CRD files should be deleted.

File: .gitignore [hunk 1]
```diff
@@ -42,3 +42,5 @@ apisix-ingress-controller
 .actions/openwhisk-utilities
 
 .generated
+
+bin/
```
Comment (lingsamuel) [inline .gitignore:46]: Missing EOF

**Comment by lingsamuel** [comment]
We can update the gitattribute file after this PR merge.

File: pkg/providers/apisix/translation/apisix_route.go [hunk 2]
```diff
@@ -136,8 +137,10 @@ func (t *translator) translateHTTPRouteV2beta2(ctx *translation.TranslateContext
 			if !plugin.Enable {
 				continue
 			}
-			if plugin.Config != nil {
-				pluginMap[plugin.Name] = plugin.Config
+			if plugin.Config.Size() > 0 {
+				config := map[string]interface{}{}
+				_ = json.Unmarshal(plugin.Config.Raw, &config)
```
Comment (lingsamuel) [inline pkg/providers/apisix/translation/apisix_route.go:142]: Why the error is ignored?

File: pkg/kube/apisix/apis/config/v1/zz_generated.register.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ import (
 )
 
 // GroupName specifies the group name used to register the objects.
-const GroupName = "apisix.apache.org"
+const GroupName = "config"
```
Comment (lingsamuel) [inline pkg/kube/apisix/apis/config/v1/zz_generated.register.go:27]: I guess this is wrong. Leave the group option empty to let it use the domain (apisix.apache.org) directly.

**Comment by tao12345666333** [comment]
Since there will be a lot of changes involved here, and there will be a lot of conflicts, I want to put it in the v1.7 version.

**Comment by tao12345666333** [comment]
This is a huge PR, but we don't have much time to continue reviewing it. Let me move it to v1.8 and hopefully we can complete it in the next release.

**Comment by Revolyssup** [comment]
Removing this from v1.8 milestone

**Pull request closed without merging**